cert/vault_client.go

```diff
@@ -23,11 +24,14 @@ type vaultClient struct {
 
 var DefaultVaultClient = &vaultClient{}
 
-func (c *vaultClient) Get() (*api.Client, error) {
+func (c *vaultClient) Get(vaultTokenFromFile string, vaultTokenPath string) (*api.Client, error) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
 	if c.client != nil {
+		if vaultTokenFromFile == "true" {
+			c.client.SetToken(getVaultTokenFromFile(vaultTokenPath))
```

Comment:
The token should only be modified if the file has been read successfully and the token isn't empty after stripping whitespace. Otherwise we should continue to use the existing token I think.

Also, when configuring the client with the initial token we support wrapped tokens, but here we don't. We can probably re-use the logic below, but it's a bit tricky.

We can reasonably expect that a wrapped token only works once, since that's the whole point of wrapping. In other words, the token stored in the file may be useless after the inital setup, so we can't blindly re-set the token here. We must do that only if the file contents changed, and do the unwrapping step again.

Reply:
For my use case, i trust that the token in the file is always a valid token, so i would want to grab the valid token every time fabio does a new check to vault, but i can see where that might be a challenge. I'll add some logic to only set the token if it has changed in the file (the current token no longer matches the token in the file), and account for if the file is blank or missing.